### PR TITLE
Show the list of parameters in expression documentations

### PR DESCRIPTION
**Pull request by D8H**
image

It's a place where the unit could be displayed.

### CODE REVIEW

File: newIDE/app/src/EventsSheet/ParameterRenderingService.js
Comment (D8H) [inline newIDE/app/src/EventsSheet/ParameterRenderingService.js:159]: It could even be "String" as the fact that it is an generalized identifier is a technical aspect.

File: newIDE/app/src/EventsSheet/ParameterFields/GenericExpressionField/ExpressionAutocompletionsDisplayer.js
Comment (AlexandreSi) [inline newIDE/app/src/EventsSheet/ParameterFields/GenericExpressionField/ExpressionAutocompletionsDisplayer.js:253]: Could these different checks be put in a dedicated function with a comment per line explaining why they are excluded?

Comment (4ian) [reply newIDE/app/src/EventsSheet/ParameterFields/GenericExpressionField/ExpressionAutocompletionsDisplayer.js:253]: Agree! We had this kind of stuff in some files related to the events based extensions editors.

**Review by 4ian — APPROVED** [review]
lgtm once comments are addressed! Will be very useful.

File: newIDE/app/src/EventsSheet/ParameterFields/GenericExpressionField/ExpressionAutocompletionsDisplayer.js
Comment (AlexandreSi) [inline newIDE/app/src/EventsSheet/ParameterFields/GenericExpressionField/ExpressionAutocompletionsDisplayer.js:254]: It might feel better organized visually if you remove the margin on the `Text` components and put all the components in a column.
Plus, you should have a warning because they don't have a key props. I think you can use the type + an index for the key?

Comment (D8H) [reply newIDE/app/src/EventsSheet/ParameterFields/GenericExpressionField/ExpressionAutocompletionsDisplayer.js:265]: I'm not sure to follow.
The screenshot of the description is not up to date. It looks like this:

image

Comment (D8H) [reply newIDE/app/src/EventsSheet/ParameterFields/GenericExpressionField/ExpressionAutocompletionsDisplayer.js:265]: @AlexandreSi do you think something need to be changed?

Comment (AlexandreSi) [reply newIDE/app/src/EventsSheet/ParameterFields/GenericExpressionField/ExpressionAutocompletionsDisplayer.js:265]: What I saw was that there is lot of space between each parameter (due to the `Text` component margins). I thought that maybe you could remove those margins (but maybe they will be too close to each other, I'll let you decide).

If, as a result, there's not enough margin between the description and the parameters, you can wrap the parameters in a `Column` component that brings its own margin.

Finally, the component that results from the map does not have a key, so React should be printing some errors/warnings in your JS console saying that they miss a key. So you could set the key such as:

```jsx
<Text style={defaultTextStyle} size="body2" key={`expression-metadata-parameter-${parameterIndex}`}>
```

Comment (D8H) [reply newIDE/app/src/EventsSheet/ParameterFields/GenericExpressionField/ExpressionAutocompletionsDisplayer.js]: Ok, I added the key.
I like the spacing, it's not far from what VSC does.

image

**Pull request merged**
